src/Servers/Connections.Abstractions/src/ConnectionContext.cs

```diff
@@ -18,6 +20,9 @@ public abstract class ConnectionContext
 
         public abstract IDuplexPipe Transport { get; set; }
 
+        public abstract EndPoint LocalEndpoint { get; set; }
+        public abstract EndPoint RemoteEndpoint { get; set; }
+
```

Comment:
If we have the Endpoints here (lgtm) then it seems we don't need `IConnectionEndPointFeature` - should we axe it?

Reply:
The features are still useful upstack where ConnectionContext isn't available.

Reply:
Generally, the ConnectionContext isn't the source of truth, features are and the implementation should be using the feature. This enables decoration of features by middleware without having to make a derived ConnectionContext